Update UA stylesheet to use `shorthands` where appropriate

Ahmad-S792 opened the pull request:
#### 97e8727d49a2beae1d2a730c83b8082358931002
<pre>
Update UA stylesheet to use `shorthands` where appropriate
<a href="https://bugs.webkit.org/show_bug.cgi?id=290605">https://bugs.webkit.org/show_bug.cgi?id=290605</a>
<a href="https://rdar.apple.com/148076358">rdar://148076358</a>

Reviewed by Tim Nguyen.

This patch is to use CSS `shorthands` where possible aligning with recent
Web Specification changes [1]:

[1] <a href="https://github.com/whatwg/html/pull/10181">https://github.com/whatwg/html/pull/10181</a>

* Source/WebCore/css/html.css:
(p):
(blockquote):
(figure):
(hr):
(dialog):
(dialog:modal):
(h1):
(:is(article, aside, nav, section) h1):
(:is(article, aside, nav, section) :is(article, aside, nav, section) h1):
(:is(article, aside, nav, section) :is(article, aside, nav, section) :is(article, aside, nav, section) h1):
(:is(article, aside, nav, section) :is(article, aside, nav, section) :is(article, aside, nav, section) :is(article, aside, nav, section) h1):
(:is(article, aside, nav, section) :is(article, aside, nav, section) :is(article, aside, nav, section) :is(article, aside, nav, section) :is(article, aside, nav, section) h1):
(h2):
(h3):
(h4):
(h5):
(h6):
(ul, menu, dir):
(ol):
(:is(dir, dl, menu, ol, ul) :is(dir, dl, menu, ol, ul)):
(dl):
(legend):
(fieldset):
(input::-webkit-credentials-auto-fill-button):
(input::-webkit-contacts-auto-fill-button):
(input::-webkit-credit-card-auto-fill-button):
(input::-internal-loading-auto-fill-button):

Canonical link: <a href="https://commits.webkit.org/292814@main">https://commits.webkit.org/292814@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/86f2c72bd92ef57686d54ce451cca96629571784

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/97195 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/16818 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/7029 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/102276 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/47720 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/99240 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/17111 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/25268 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/74044 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/31244 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/100198 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/12927 "Found 1 new test failure: fast/forms/ios/focus-input-via-button.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/87920 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54390 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/12681 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/5772 "Passed tests") | [❌ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/47126 "Failed to checkout and rebase branch from PR 43171") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/82726 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/5849 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/104298 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/24271 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/17705 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/83084 "Passed tests") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/24646 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/84023 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/82500 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20750 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/4726 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/17774 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/24234 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/29385 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/24057 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/27369 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/25630 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -186,8 +178,7 @@ h1 {
     font-size: 2em;
     margin-block-start: 0.67__qem;
     margin-block-end: 0.67em;
```
```suggestion
    margin-block: 0.67__qem 0.67em;
```

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -226,8 +217,7 @@ h2 {
     font-size: 1.5em;
     margin-block-start: 0.83__qem;
     margin-block-end: 0.83em;
```
```suggestion
    margin-block: 0.83__qem 0.83em;
```

Ahmad-S792 replied:
I didn't do it because of `__qem` (quirky em) but happy to do.

nt1m replied:
I would do it since it's more consistent that way

Ahmad-S792 replied:
Yep - just doing changes locally, will push it shortly. :-)

Ahmad-S792 replied:
@nt1m - done for all possible cases IMO. :-)

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -236,17 +226,15 @@ h3 {
     font-size: 1.17em;
     margin-block-start: 1__qem;
     margin-block-end: 1em;
```
ditto

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -236,17 +226,15 @@ h3 {
 h4 {
     display: block;
     margin-block-start: 1.33__qem;
     margin-block-end: 1.33em;
```
ditto

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -255,8 +243,7 @@ h5 {
     font-size: .83em;
     margin-block-start: 1.67__qem;
     margin-block-end: 1.67em;
```
ditto

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -265,8 +252,7 @@ h6 {
     font-size: .67em;
     margin-block-start: 2.33__qem;
     margin-block-end: 2.33em;
```
ditto

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -62,8 +61,7 @@ blockquote {
     display: block;
     margin-block-start: 1__qem;
     margin-block-end: 1em;
```
ditto: `margin-block: 1__qem 1em;`

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -45,8 +45,7 @@ p {
     display: block;
     margin-block-start: 1__qem;
     margin-block-end: 1__qem;
```
```suggestion
    margin-block: 1__qem;
```

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -43,10 +43,8 @@ body {
-    margin-block-end: 1__qem;
-    margin-inline-start: 0;
-    margin-inline-end: 0;
+    margin-block: 1__qem 1__qem;
```
```suggestion
    margin-block: 1__qem;
```

Ahmad-S792 replied:
Done.

Ahmad-S792 committed:
Update UA stylesheet to use `shorthands` where appropriate
https://bugs.webkit.org/show_bug.cgi?id=290605
rdar://148076358

Reviewed by Tim Nguyen.

This patch is to use CSS `shorthands` where possible aligning with recent
Web Specification changes [1]:

[1] https://github.com/whatwg/html/pull/10181

* Source/WebCore/css/html.css:
(p):
(blockquote):
(figure):
(hr):
(dialog):
(dialog:modal):
(h1):
(:is(article, aside, nav, section) h1):
(:is(article, aside, nav, section) :is(article, aside, nav, section) h1):
(:is(article, aside, nav, section) :is(article, aside, nav, section) :is(article, aside, nav, section) h1):
(:is(article, aside, nav, section) :is(article, aside, nav, section) :is(article, aside, nav, section) :is(article, aside, nav, section) h1):
(:is(article, aside, nav, section) :is(article, aside, nav, section) :is(article, aside, nav, section) :is(article, aside, nav, section) :is(article, aside, nav, section) h1):
(h2):
(h3):
(h4):
(h5):
(h6):
(ul, menu, dir):
(ol):
(:is(dir, dl, menu, ol, ul) :is(dir, dl, menu, ol, ul)):
(dl):
(legend):
(fieldset):
(input::-webkit-credentials-auto-fill-button):
(input::-webkit-contacts-auto-fill-button):
(input::-webkit-credit-card-auto-fill-button):
(input::-internal-loading-auto-fill-button):

Canonical link: https://commits.webkit.org/292814@main
